src/main/kotlin/no/nav/familie/ef/sak/vedtak/VedtakHistorikkBeregner.kt

```diff
@@ -17,24 +19,30 @@ object VedtakHistorikkBeregner {
 
     private fun lagTotalbildeForNyttVedtak(vedtak: Vedtak,
                                            acc: List<Pair<UUID, List<Vedtaksperiode>>>): List<Vedtaksperiode> {
-        val nyePerioder = vedtak.perioder!!.perioder
-        return avkortTidligerePerioder(acc.lastOrNull(), nyePerioder.first()) + nyePerioder
+        return if (vedtak.resultatType == ResultatType.INNVILGE) {
+            val nyePerioder = vedtak.perioder?.perioder ?: error("Finner ikke perioder på vedtak=${vedtak.behandlingId}")
+            val førsteFraDato = nyePerioder.first().datoFra
+            avkortTidligerePerioder(acc.lastOrNull(), førsteFraDato) + nyePerioder
+        } else {
+            val opphørFom = vedtak.opphørFom ?: error("Mangler dato for opphør på vedtak=${vedtak.behandlingId}")
+            avkortTidligerePerioder(acc.lastOrNull(), opphørFom)
```

Comment:
Kan det skje at man skal "avkorte" flere perioder?

Reply:
Muntlig gjennomgang av testcase ~
lastOrNull her er på siste totalbildet for vedtaksperioder